--- zippy.c	2003-11-25 00:25:20.000000000 -0500
+++ zippy.c.new	2008-06-28 21:40:47.000000000 -0400
@@ -503,9 +503,11 @@
 	} else if (appData.zippyPassword2[0] != NULLCHAR && first.initDone &&
 	    strncmp(star_match[1], appData.zippyPassword2,
 		    strlen(appData.zippyPassword2)) == 0) {
+	    char zp2buf[1024];
 	    p = star_match[1] + strlen(appData.zippyPassword2);
 	    while (*p == ' ') p++;
-	    SendToProgram(p, &first);
+	    snprintf(zp2buf, 1023, "zippy %s %s", player, p);
+	    SendToProgram(zp2buf, &first);
 	    SendToProgram("\n", &first);
 	} else if (appData.zippyWrongPassword[0] != NULLCHAR &&
 	    strncmp(star_match[1], appData.zippyWrongPassword,
